catalog/core/catalog-core-validator/src/main/java/ddf/catalog/validation/impl/validator/RelationshipValidator.java

```diff
@@ -0,0 +1,257 @@
+          "Unrecognized relationship " + relationship + " for validator " + this.toString());
+    }
+    if (CAN_ONLY_HAVE.equals(relationship) && CollectionUtils.isEmpty(this.targetValues)) {
+      throw new IllegalArgumentException(
+          "Relationship " + CAN_ONLY_HAVE + " must specify one or more target values");
+    }
+  }
+
+  @Override
+  public Optional<MetacardValidationReport> validateMetacard(Metacard metacard) {
+    Attribute attribute = metacard.getAttribute(sourceAttribute);
+    if (attribute != null
+        && attribute.getValues().stream().anyMatch(Objects::nonNull)
+        && (StringUtils.isEmpty(sourceValue) || attribute.getValues().contains(sourceValue))) {
+      MetacardValidationReportImpl report = new MetacardValidationReportImpl();
+      Optional<ValidationViolation> violation =
+          relationships.get(relationship).apply(metacard.getAttribute(targetAttribute));
+      violation.ifPresent(report::addAttributeViolation);
+      if (violation.isPresent()) {
+        return Optional.of(report);
+      } else {
+        return Optional.empty();
+      }
+    } else {
+      return Optional.empty();
+    }
+  }
+
+  private Optional<ValidationViolation> mustHave(Attribute attribute) {
+    return getValidationViolation(
+        "must have",
+        attribute == null
+            || (targetValues.isEmpty()
+                ? attribute
+                        .getValues()
+                        .stream()
+                        .map(Objects::toString)
+                        .filter(StringUtils::isNotEmpty)
+                        .count()
+                    < 1
+                : !attribute
+                    .getValues()
+                    .stream()
+                    .map(Objects::toString)
+                    .filter(StringUtils::isNotEmpty)
+                    .collect(Collectors.toCollection(ArrayList::new))
+                    .containsAll(targetValues)));
+  }
+
+  private Optional<ValidationViolation> cannotHave(Attribute attribute) {
+    return getValidationViolation(
+        "cannot have",
+        attribute != null
+            && attribute
+                .getValues()
+                .stream()
+                .map(Objects::toString)
+                .filter(StringUtils::isNotEmpty)
+                .anyMatch(v -> targetValues.isEmpty() || targetValues.contains(v)));
+  }
+
+  private Optional<ValidationViolation> canOnlyHave(Attribute attribute) {
+    return getValidationViolation(
+        "can only have",
+        attribute != null
+            && !CollectionUtils.isEmpty(attribute.getValues())
+            && !attribute
+                .getValues()
+                .stream()
+                .map(Objects::toString)
+                .filter(StringUtils::isNotEmpty)
+                .allMatch(targetValues::contains));
+  }
+
+  private String getValuesAsString() {
+    return targetValues.stream().map(Object::toString).collect(Collectors.joining(", "));
```

Comment:
✏️ `targetValues` is a `List<String>` so we don't need to use `toString`. Could use `String.join(", ", targetValues)` instead.